src/unlock/omni_lock.rs

```diff
@@ -241,6 +241,55 @@ impl PartialEq for SmtProofEntryVec {
 }
 impl Eq for SmtProofEntryVec {}
 
+pub struct InfoCellData {
+    pub version: u8,
+    pub current_supply: u128,
+    pub max_supply: u128,
+    pub sudt_script_hash: H256,
+    pub other_data: Vec<u8>,
+}
+
+impl InfoCellData {
+    pub fn new_simple(current_supply: u128, max_supply: u128) -> Self {
+        InfoCellData::new(current_supply, max_supply, H256::default(), vec![])
+    }
```

Comment:
From [omni lock rfc](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0042-omnilock/0042-omnilock.md#supply-mode):
> When supply mode is enabled, <32 bytes type script hash> must be present.

So, this method should be removed, and rename `new_with_script_hash` to `new_simple`.